server/src/main/java/org/spine3/server/projection/ProjectionRepository.java

```diff
@@ -287,21 +323,27 @@ public void setOnline() {
         setStatus(Status.ONLINE);
     }
 
-   /**
+    /**
      * The stream observer which redirects events from {@code EventStore} to
      * the associated {@code ProjectionRepository}.
      */
     private static class EventStreamObserver implements StreamObserver<Event> {
 
         private final ProjectionRepository projectionRepository;
+        private BulkWriteOperation operation;
 
         EventStreamObserver(ProjectionRepository projectionRepository) {
             this.projectionRepository = projectionRepository;
         }
 
         @Override
         public void onNext(Event event) {
+            if (operation == null || operation.isInProgress()) {
```

Comment:
This line doesn't seem to be checked in full. And, BTW, it's the same condition check as above. Why don't we have a method then?